cmd/exp/entropy.go

```diff
@@ -0,0 +1,100 @@
+package exp
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-logfmt/logfmt"
+	"github.com/hashicorp/terraform/plans"
+	"github.com/hashicorp/terraform/plans/planfile"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	entropyCmd.Flags().StringP("plan-file", "f", "TODO", "Path to Terraform Plan file to parse.")
+	entropyCmd.Flags().StringP("output-file", "o", "TODO", "Path to write instrumentation to")
+	entropyCmd.Flags().StringP("component", "c", "TODO", "The terraform component for this plan")
+	entropyCmd.Flags().StringP("project", "p", "TODO", "The terraform project for this plan")
+
+	entropyCmd.MarkFlagRequired("component")
+	entropyCmd.MarkFlagRequired("project")
+
+	ExpCmd.AddCommand(entropyCmd)
+}
+
+type terraformDiff struct {
+	Address      string `logfmt:"address,omitempty"`
+	ResourceMode string `logfmt:"resource_mode,omitempty"`
+	Action       string `logfmt:"action,omitempty"`
+	Project      string `logfmt:"project,omitempty"`
+	Component    string `logfmt:"component,omitempty"`
+}
+
+var entropyCmd = &cobra.Command{
+	Use:   "entropy",
+	Short: "Measures how many differences result from a terraform plan.",
+	Long: `This command will parse a Terraform plan and track any diffs.
+It is meant to be run with honeycomb/buildevents and thus we generate
+output in LogFmt format.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		planFilePath, err := cmd.Flags().GetString("plan-file")
+		if err != nil {
+			return fmt.Errorf("could not read plan-file flag: %w", err)
+		}
+		outputFilePath, err := cmd.Flags().GetString("output-file")
+		if err != nil {
+			return fmt.Errorf("could not read output-file flag: %w", err)
+		}
+		component, err := cmd.Flags().GetString("component")
+		if err != nil {
+			return fmt.Errorf("could not read component flag: %w", err)
+		}
+		project, err := cmd.Flags().GetString("project")
+		if err != nil {
+			return fmt.Errorf("could not read project flag: %w", err)
+		}
+
+		planReader, err := planfile.Open(planFilePath)
+		if err != nil {
+			return fmt.Errorf("could not open terraform plan: %w", err)
+		}
+		defer planReader.Close()
+
+		plan, err := planReader.ReadPlan()
+		if err != nil {
+			return fmt.Errorf("could not read/parse terraform plan: %w", err)
+		}
+
+		f, err := os.Create(outputFilePath)
+		if err != nil {
+			return fmt.Errorf("could not open output file for writing: %w", err)
+		}
+		defer f.Close()
+
+		encoder := logfmt.NewEncoder(f)
+		for _, resourceChange := range plan.Changes.Resources {
+			addr := resourceChange.Addr.String()
+			resourceMode := resourceChange.Addr.Resource.Resource.Mode.String()
+			action := resourceChange.Action
+
+			if action == plans.NoOp {
+				continue
+			}
+
+			err = encoder.EncodeKeyvals(
+				"addr", addr,
+				"resource_mode", resourceMode,
+				"action", action,
+				"component", component,
+				"project", project)
+			if err != nil {
+				return fmt.Errorf("could not encode key/val: %w", err)
+			}
+			err = encoder.EndRecord()
```

Comment:
Have you tested that having one k/v line works? The docs seem to indicate yes.

Reply:
I;ve yet to test this in conjunction with buildevents. Wanted to merge the buildevents pr first